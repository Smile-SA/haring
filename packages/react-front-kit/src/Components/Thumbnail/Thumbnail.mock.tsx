import type { IThumbnailAction } from './Thumbnail';

import {
  DownloadSimple,
  Eye,
  PencilSimple,
  ShareNetwork,
  Star,
  Trash,
} from '@phosphor-icons/react';
import { FolderMove } from '@smile/react-front-kit-shared';
import { action } from '@storybook/addon-actions';

export const thumbnailActions: IThumbnailAction[] = [
  {
    icon: <FolderMove />,
    id: 'move',
    label: 'Move in tree',
    onAction: action('Move in tree'),
  },
  {
    icon: <Eye />,
    id: 'open',
    label: 'Open document',
    onAction: action('Open document'),
  },
  {
    icon: <PencilSimple />,
    id: 'edit',
    label: 'Edit document',
    onAction: action('Edit document'),
  },
  {
    icon: <Star />,
    id: 'favorite',
    label: 'Add to favorites',
    onAction: action('Add to favorites'),
  },
  {
    icon: <ShareNetwork />,
    id: 'share',
    label: 'Share',
    onAction: action('Share'),
  },
  {
    icon: <DownloadSimple />,
    id: 'download',
    label: 'Download',
    onAction: action('Download'),
  },
  {
    color: 'red',
    confirmModalProps: {
      cancelLabel: 'Abort',
      children: <p>Are you sure ?</p>,
      confirmColor: 'red',
      confirmLabel: 'Remove',
      title: 'Remove File',
    },
    confirmation: true,
    icon: <Trash />,
    id: 'delete',
    label: 'Delete',
    onAction: action('Delete'),
  },
];
