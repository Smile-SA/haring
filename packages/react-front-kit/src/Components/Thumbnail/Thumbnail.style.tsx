import { createStyles } from '@mantine/styles';

export const useStyles = createStyles((theme) => ({
  dotsIcon: {
    color: theme.colors.cyan[9],
  },
  headerContainer: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'space-between',
    marginBottom: '22px',
    marginTop: '10px',
  },
  icon: {
    height: 'fit-content',
    margin: 'auto',
  },
  iconTitle: {
    minWidth: '28px',
  },
  menuButton: {
    '&[aria-expanded=true]': {
      '& svg': {
        filter: 'brightness(0) invert(1)',
      },
      backgroundColor: theme.colors.cyan[9],
      borderRadius: '4px',
      display: 'flex',
      height: '28px',
      width: '28px',
    },
  },
  menuButtonSelected: {
    '& svg': {
      filter: 'brightness(0) invert(1)',
    },
    '&[aria-expanded=true]': {
      '& svg': {
        filter: 'none',
      },
      backgroundColor: theme.colors.cyan[0],
    },
    ':hover': {
      '& svg': {
        filter: 'none',
      },
      backgroundColor: theme.colors.cyan[1],
    },
  },
  root: {
    background: theme.colors.gray[1],
    borderRadius: '16px',
    heigh: 'auto',
    padding: '16px',
    width: 'auto',
  },
  rootSelected: { background: theme.primaryColor },
  title: {
    margin: '0 19px 0 0',
    paddingLeft: '10px',
  },
  titleContainer: {
    alignItems: 'center',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden',
  },
  titleContainerRootSelected: {
    color: 'white',
  },
}));
