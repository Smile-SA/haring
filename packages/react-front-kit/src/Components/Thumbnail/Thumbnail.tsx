'use client';

import type { IThumbnail, IThumbnailAction } from '../../types';
import type { IActionConfirmModalProps } from '@smile/haring-react-shared';
import type { ReactElement } from 'react';

import {
  ActionIcon,
  Box,
  Group,
  Image,
  Menu,
  Text,
  getThemeColor,
  useMantineTheme,
} from '@mantine/core';
import { DotsThreeVertical } from '@phosphor-icons/react';
import { FileIcon } from '@smile/haring-react-shared';
import { useState } from 'react';

import defaultImage from '../../../assets/defaultImage.jpg';
import { ConfirmModal } from '../ConfirmModal/ConfirmModal';

import classes from './Thumbnail.module.css';

export interface IThumbnailProps extends IThumbnail {
  actions?: IThumbnailAction[];
}

export function Thumbnail(props: IThumbnailProps): ReactElement {
  const theme = useMantineTheme();
  const {
    actions = [],
    iconType,
    image = defaultImage,
    label,
    onClick,
    selected = false,
  } = props;
  const [confirmAction, setConfirmAction] =
    useState<IActionConfirmModalProps<IThumbnail> | null>(null);

  function clearConfirmAction(): void {
    setConfirmAction(null);
  }

  const rootClasses = [classes.root];
  if (selected) {
    rootClasses.push(classes.rootSelected);
  }

  function setModal(action: IThumbnailAction): void {
    setConfirmAction({
      cancelColor: action.confirmModalProps?.cancelColor,
      cancelLabel: action.confirmModalProps?.cancelLabel,
      children: action.confirmModalProps?.children,
      confirmColor: action.confirmModalProps?.confirmColor,
      confirmLabel: action.confirmModalProps?.confirmLabel,
      onConfirm: () => action.onAction?.(props),
      title: action.confirmModalProps?.title,
    });
  }

  function handleMenuItem(action: IThumbnailAction): void {
    if (action.confirmation) {
      setModal(action);
    } else {
      action.onAction?.(props);
    }
  }

  function handleClose(): void {
    clearConfirmAction();
  }

  function handleModalButton(onModalAction?: (item: IThumbnail) => void): void {
    onModalAction?.(props);
    handleClose();
  }

  return (
    <>
      <Box
        bg={String(
          selected
            ? getThemeColor(theme.primaryColor, theme)
            : theme.colors.gray[1],
        )}
        className={rootClasses.join(' ')}
        onClick={onClick}
      >
        <Group className={classes.headerContainer}>
          <div className={classes.titleContainer}>
            <FileIcon
              className={classes.fileIcon}
              color={String(
                selected
                  ? theme.colors.gray[1]
                  : getThemeColor(theme.primaryColor, theme),
              )}
              size={22}
              type={iconType}
              weight="bold"
            />
            <Text className={classes.title} component="h3" truncate>
              {label}
            </Text>
          </div>
          <div>
            {actions.length > 0 && (
              <Menu radius={4} shadow="lg" width={200}>
                <Menu.Target>
                  <ActionIcon
                    className={
                      selected ? classes.menuButtonSelected : classes.menuButton
                    }
                    onClick={(e) => e.stopPropagation()}
                    radius={4}
                    type="button"
                  >
                    <div>
                      <DotsThreeVertical
                        className={classes.dotsIcon}
                        size={16}
                        weight="bold"
                      />
                    </div>
                  </ActionIcon>
                </Menu.Target>
                <Menu.Dropdown onClick={(e) => e.stopPropagation()}>
                  {actions.map((action, index) => (
                    <Menu.Item
                      // eslint-disable-next-line react/no-array-index-key
                      key={index}
                      color={action.color}
                      leftSection={
                        typeof action.icon === 'function'
                          ? action.icon(props)
                          : action.icon
                      }
                      onClick={() => handleMenuItem(action)}
                      {...(typeof action.componentProps === 'function'
                        ? action.componentProps(props)
                        : action.componentProps)}
                    >
                      {typeof action.label === 'function'
                        ? action.label(props)
                        : action.label}
                    </Menu.Item>
                  ))}
                </Menu.Dropdown>
              </Menu>
            )}
          </div>
        </Group>
        <Image radius="16px" src={image} />
      </Box>
      <ConfirmModal
        {...confirmAction}
        onCancel={() => handleModalButton(confirmAction?.onCancel)}
        onClose={handleClose}
        onConfirm={() => handleModalButton(confirmAction?.onConfirm)}
        opened={Boolean(confirmAction)}
      >
        {confirmAction?.children}
      </ConfirmModal>
    </>
  );
}
