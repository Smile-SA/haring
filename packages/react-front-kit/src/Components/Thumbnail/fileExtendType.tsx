import type { MantineColor } from '@mantine/core';
import type { ReactElement } from 'react';

import {
  File,
  FileCss,
  FileCsv,
  FileDoc,
  FileHtml,
  FileJpg,
  FileJs,
  FileJsx,
  FilePdf,
  FilePng,
  FilePpt,
  FileRs,
  FileSql,
  FileSvg,
  FileTs,
  FileTsx,
  FileVue,
  FileXls,
  FileZip,
} from '@phosphor-icons/react';

export type IFileExtendType =
  | 'CSS'
  | 'CSV'
  | 'DOC'
  | 'HTML'
  | 'JPG'
  | 'JS'
  | 'JSX'
  | 'PDF'
  | 'PNG'
  | 'PPT'
  | 'RS'
  | 'SQL'
  | 'SVG'
  | 'TS'
  | 'TSX'
  | 'UNKNOWN'
  | 'VUE'
  | 'XLS'
  | 'ZIP';

export const getIconByIconType = (
  type: IFileExtendType,
  color: MantineColor,
): ReactElement => {
  switch (type) {
    case 'CSS':
      return (
        <FileCss
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'DOC':
      return (
        <FileDoc
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'PDF':
      return (
        <FilePdf
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'CSV':
      return (
        <FileCsv
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'HTML':
      return (
        <FileHtml
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'JPG':
      return (
        <FileJpg
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'JS':
      return (
        <FileJs
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'JSX':
      return (
        <FileJsx
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'PNG':
      return (
        <FilePng
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'PPT':
      return (
        <FilePpt
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'RS':
      return (
        <FileRs
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'SQL':
      return (
        <FileSql
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'SVG':
      return (
        <FileSvg
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'TS':
      return (
        <FileTs
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'TSX':
      return (
        <FileTsx
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'VUE':
      return (
        <FileVue
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'XLS':
      return (
        <FileXls
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    case 'ZIP':
      return (
        <FileZip
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    default: {
      return (
        <File
          color={color}
          size={22}
          style={{ minWidth: '28px' }}
          weight="bold"
        />
      );
    }
  }
};
