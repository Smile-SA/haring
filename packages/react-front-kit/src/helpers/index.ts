export * from './nestedObject';
export * from './typeGuard';
