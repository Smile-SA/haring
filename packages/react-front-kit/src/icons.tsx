import type { Icon } from '@phosphor-icons/react';

import * as Icons from '@phosphor-icons/react';

const iconsToIgnore = ['IconBase', 'IconContext'];
const iconEntries = Object.entries(Icons).filter(
  ([key]) => !iconsToIgnore.includes(key)
) as [string, Icon][];

export const icons = Object.fromEntries(iconEntries);

export const iconsElements = Object.fromEntries(
  iconEntries.map(([key, Icon]) => [key, <Icon key={key} />])
);
