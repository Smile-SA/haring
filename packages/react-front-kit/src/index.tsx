/* eslint-disable react-refresh/only-export-components */
'use client';

// component exports
export type { ICardHeaderProps } from './Components/CardHeader/CardHeader';
export { CardHeader } from './Components/CardHeader/CardHeader';
export type { IMenuLanguagesProps } from './Components/MenuLanguages/MenuLanguages';
export { MenuLanguages } from './Components/MenuLanguages/MenuLanguages';
export type { IButtonListOrDropdownProps } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ButtonListOrDropdown } from './Components/ButtonListOrDropdown/ButtonListOrDropdown';
export { ActionBar } from './Components/ActionBar/ActionBar';
export type { IActionBarProps } from './Components/ActionBar/ActionBar';
export { ActionRowOverflow } from './Components/ActionRowOverflow/ActionRowOverflow';
export type {
  IActionRowOverflowProps,
  IActionRowOverflowAction,
} from './Components/ActionRowOverflow/ActionRowOverflow';
export { BitConverter } from './Components/BitConverter/BitConverter';
export type { IBitConverterProps } from './Components/BitConverter/BitConverter';
export { Breadcrumbs } from './Components/Breadcrumbs/Breadcrumbs';
export type { IBreadcrumbsProps } from './Components/Breadcrumbs/Breadcrumbs';
export { CollapseButtonControlled } from './Components/CollapseButton/CollapseButtonControlled';
export type { ICollapseButtonControlledProps } from './Components/CollapseButton/CollapseButtonControlled';
export { CollapseButton } from './Components/CollapseButton/CollapseButton';
export type { ICollapseButtonProps } from './Components/CollapseButton/CollapseButton';
export { ConfirmModal } from './Components/ConfirmModal/ConfirmModal';
export type { IConfirmModalProps } from './Components/ConfirmModal/ConfirmModal';
export { DocumentBox } from './Components/DocumentBox/DocumentBox';
export type { IDocumentBoxProps } from './Components/DocumentBox/DocumentBox';
export { DocumentList } from './Components/DocumentList/DocumentList';
export type {
  IDocument,
  IDocumentListProps,
} from './Components/DocumentList/DocumentList';
export { DropdownButton } from './Components/DropdownButton/DropdownButton';
export type { IDropdownButtonProps } from './Components/DropdownButton/DropdownButton';
export { FilterList } from './Components/FilterList/FilterList';
export type { IFilterListProps } from './Components/FilterList/FilterList';
export { Header } from './Components/Header/Header';
export type { IHeaderProps } from './Components/Header/Header';
export { HeaderMobile } from './Components/HeaderMobile/HeaderMobile';
export type { IHeaderMobileProps } from './Components/HeaderMobile/HeaderMobile';
export { HeaderNav } from './Components/HeaderNav/HeaderNav';
export type {
  IHeaderNavMenu,
  IHeaderNavProps,
} from './Components/HeaderNav/HeaderNav';
export { HeaderSearch } from './Components/HeaderSearch/HeaderSearch';
export type { IHeaderSearchProps } from './Components/HeaderSearch/HeaderSearch';
export { InfoCard } from './Components/InfoCard/InfoCard';
export type {
  IContentItem,
  IInfoCardProps,
} from './Components/InfoCard/InfoCard';
export { Motif } from './Components/InfoCard/Motif';
export type { IMotifProps } from './Components/InfoCard/Motif';
export { Pagination } from './Components/Pagination/Pagination';
export type { IPaginationProps } from './Components/Pagination/Pagination';
export { Preview } from './Components/Preview/Preview';
export type { IPreviewProps } from './Components/Preview/Preview';
export { ResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export type { IResponsiveTabs } from './Components/ResponsiveTabs/ResponsiveTabs';
export { SearchableCheckboxList } from './Components/SearchableCheckboxList/SearchableCheckboxList';
export type {
  ISearchableCheckboxListProps,
  ICheckbox,
} from './Components/SearchableCheckboxList/SearchableCheckboxList';
export { SearchBar } from './Components/SearchBar/SearchBar';
export type { ISearchBarProps } from './Components/SearchBar/SearchBar';
export { SelectableList } from './Components/SelectableList/SelectableList';
export type { ISelectableListProps } from './Components/SelectableList/SelectableList';
export { SidebarFilters } from './Components/SidebarFilters/SidebarFilters';
export type {
  ISidebarFiltersProps,
  ISidebarFilter,
} from './Components/SidebarFilters/SidebarFilters';
export {
  getRecursiveMenu,
  SidebarMenu,
} from './Components/SidebarMenu/SidebarMenu';
export type {
  IMenuItem,
  ISidebarMenuProps,
} from './Components/SidebarMenu/SidebarMenu';
export { SummaryBox } from './Components/SummaryBox/SummaryBox';
export type { ISummaryBoxProps } from './Components/SummaryBox/SummaryBox';
export { SwitchableView } from './Components/SwitchableView/SwitchableView';
export type {
  IDataView,
  ISwitchableViewProps,
} from './Components/SwitchableView/SwitchableView';
export { Thumbnail } from './Components/Thumbnail/Thumbnail';
export type { IThumbnailProps } from './Components/Thumbnail/Thumbnail';
export { ThumbnailGrid } from './Components/ThumbnailGrid/ThumbnailGrid';
export type { IThumbnailGridProps } from './Components/ThumbnailGrid/ThumbnailGrid';
// layout exports
export { FoldableColumnLayout } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
export type { IFoldableColumnLayoutProps } from './Layouts/FoldableColumnLayout/FoldableColumnLayout';
// helper exports
export {
  addPathAndDepth,
  flattenNestedObjects,
  getClosestItemFromPath,
  setChildrenToTree,
} from './helpers';
// type exports
export type { IThumbnail, IThumbnailAction, IThumbnailData } from './types';
