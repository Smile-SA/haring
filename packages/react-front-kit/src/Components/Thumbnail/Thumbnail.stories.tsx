import type { Meta, StoryObj } from '@storybook/react';

import { Thumbnail as Cmp } from './Thumbnail';
import { thumbnailActionsMock } from './Thumbnail.mock';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/Thumbnail',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const Thumbnail: IStory = {
  args: {
    actions: thumbnailActionsMock,
    iconType: 'PDF',
    id: '1',
    label: 'Debit_Suivi_PREV',
    selected: false,
  },
};
