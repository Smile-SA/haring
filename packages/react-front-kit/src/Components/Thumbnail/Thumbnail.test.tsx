import { renderWithProviders } from '@smile/react-front-kit-shared/src/test-utils';

import { Thumbnail } from './Thumbnail';

describe('Thumbnail', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<Thumbnail />);
    expect(container).toMatchSnapshot();
  });
});
