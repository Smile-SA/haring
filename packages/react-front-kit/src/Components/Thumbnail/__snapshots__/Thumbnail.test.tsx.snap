// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail matches snapshot 1`] = `
<div>
  <div
    class="mantine-af86sk"
    id="thumbnail-test"
  >
    <div
      class="mantine-Group-root mantine-p1wipv"
    >
      <div
        class="mantine-1r83wdq"
      >
        <svg
          fill="#0b7285"
          height="22"
          viewBox="0 0 256 256"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M216.49,79.52l-56-56A12,12,0,0,0,152,20H56A20,20,0,0,0,36,40V216a20,20,0,0,0,20,20H200a20,20,0,0,0,20-20V88A12,12,0,0,0,216.49,79.52ZM160,57l23,23H160ZM60,212V44h76V92a12,12,0,0,0,12,12h48V212Z"
          />
        </svg>
        <h3
          class="mantine-Text-root mantine-1oer4ck"
        />
      </div>
      <div />
    </div>
    <div
      class="mantine-Image-root mantine-yxmaw9"
      style="width: 100%;"
    >
      <figure
        class="mantine-qenwvq mantine-Image-figure"
      >
        <div
          class="mantine-1iugybl mantine-Image-imageWrapper"
        >
          <img
            class="mantine-1bi35m7 mantine-Image-image"
            src=""
            style="object-fit: cover; width: 100%; height: auto;"
          />
        </div>
      </figure>
    </div>
  </div>
</div>
`;
